### ISSUE TITLE
Android 8.0 CoreReceiver.onNotificationOpened crash

### ISSUE THREAD

**Issue by aechap02** [urbanairship/android-library#58]
We are receiving this crash on Android 8, which I believe is related to background execution limit changes:
https://developer.android.com/about/versions/oreo/android-8.0-changes.html#back-all

```
Caused by java.lang.IllegalStateException: Not allowed to start service Intent { act=com.urbanairship.actionservice.ACTION_RUN_ACTIONS cmp=com.cbsinteractive.cnet/com.urbanairship.actions.ActionService (has extras) }: app is in background uid UidRecord{e2ad803 u0a205 RCVR idle change:uncached procs:1 seq(0,0,0)}
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1521)
       at android.app.ContextImpl.startService(ContextImpl.java:1477)
       at android.content.ContextWrapper.startService(ContextWrapper.java:650)
       at android.content.ContextWrapper.startService(ContextWrapper.java:650)
       at com.urbanairship.actions.ActionService.runActions(ActionService.java:159)
       at com.urbanairship.CoreReceiver.onNotificationOpened(CoreReceiver.java:254)
       at com.urbanairship.CoreReceiver.onReceive(CoreReceiver.java:57)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3187)
       at android.app.ActivityThread.-wrap17(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6494)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by marc-scig** [comment]
Hi there,

Thanks for the report. What version of the SDK are you running? We did some work in recent releases refactoring the old service logic into Jobs but but it's possible we might have missed something.

**Comment by aechap02** [comment]
We are currently using 8.9.6

**Comment by aechap02** [comment]
Thank you!

**Comment by sidsharan** [comment]
Hi @marc-scig .. any update on this? I am facing the same issue in version 9.0.0

**Comment by rlepinski** [comment]
Have a fix ready. Hopefully will be released today if testing goes well.

**Comment by rlepinski** [comment]
FIxed in 9.0.2
